tidb_partitioned_tables_guide.md

```diff
@@ -0,0 +1,707 @@
+)
+ENGINE=InnoDB DEFAULT CHARSET=utf8mb4 COLLATE=utf8mb4_bin
+TTL=`expire_time` + INTERVAL 0 DAY TTL_ENABLE='ON'
+TTL_JOB_INTERVAL='10m'
+```
+
+**Drop Partition (Range INTERVAL partitioning)**
+
+```sql
+CREATE TABLE `ad_cache` (
+  `session_id` varchar(255) NOT NULL,
+  `external_id` varbinary(255) NOT NULL,
+  `create_time` datetime NOT NULL DEFAULT CURRENT_TIMESTAMP,
+  `id_suffix` bigint(20) NOT NULL,
+  `expire_time` timestamp NULL DEFAULT NULL,
+  `cache_data` mediumblob DEFAULT NULL,
+  `data_version` int(11) DEFAULT NULL,
+  `is_deleted` tinyint(1) DEFAULT NULL,
+  PRIMARY KEY (
+    `session_id`, `external_id`,
+    `create_time`, `id_suffix`
+  ) NONCLUSTERED
+)
+SHARD_ROW_ID_BITS=7
+PRE_SPLIT_REGIONS=2
+PARTITION BY RANGE COLUMNS (create_time)
+INTERVAL (10 MINUTE)
+FIRST PARTITION LESS THAN ('2025-02-19 18:00:00')
+...
+LAST PARTITION LESS THAN ('2025-02-19 20:00:00')
+```
+
+It's required to run DDL alter table partition ... to change the FIRST PARTITION and LAST PARTITION periodically. These two DDL statements can drop the old partitions and create new ones.
+
+```sql
+ALTER TABLE ad_cache FIRST PARTITION LESS THAN ("${nextTimestamp}")
+ALTER TABLE ad_cache LAST PARTITION LESS THAN ("${nextTimestamp}")
+```
+
+#### Recommendation
+
+For workloads with **large or time-based data cleanup**, prefer using **partitioned tables with DROP PARTITION**. It offers better performance, lower system impact, and simpler management. TTL is still useful for finer-grained or background cleanup but may not be optimal under high write pressure or when deleting large volumes of data quickly.
+
+### Partition Drop Efficiency: Local Index vs Global Index
+
+Partition table with Global Index requires synchronous updates to the global index, potentially increasing significant execution time for DDL operations, such as DROP PARTITION, TRUNCATE PARTITION, or REORG PARTITION. In this section, the tests show that DROP PARTITION is much slower when using a **Global Index** compared to a **Local Index**. This should be considered when designing partitioned tables.
+
+#### What Did We Test
+
+We created a table with **366 partitions** and tested the DROP PARTITION performance using both **Global Index** and **Local Index**. The total number of rows was **1 billion**.
+
+| Index Type | Duration (drop partition) |
+|---|---|
+| Global Index | 1 min 16.02 s |
+| Local Index | 0.52 s |
+
+#### Findings
+
+Dropping a partition on a table with a Global Index took **76 seconds**, while the same operation with a Local Index took only **0.52 seconds**. The reason is that Global Indexes span all partitions and require more complex updates, while Local Indexes are limited to individual partitions and are easier to handle.
+
+**Global Index**
+
+```sql
+mysql> alter table A drop partition A_2024363;
+Query OK, 0 rows affected (1 min 16.02 sec)
+```
+
+**Local Index**
+
+```sql
+mysql> alter table A drop partition A_2024363;
```

Comment:
```suggestion
When a partitioned table contains global indexes, performing certain DDL operations such as DROP PARTITION, TRUNCATE PARTITION, or REORG PARTITION requires synchronously updating the global index values. This can significantly increase the execution time of these DDL operations.
```